storage/engine/mvcc.go

```diff
@@ -1227,6 +1250,38 @@ func MVCCConditionalPut(
 	iter := engine.NewIterator(key)
 	defer iter.Close()
 
+	return mvccConditionalPutUsingIter(ctx, engine, iter, ms, key, timestamp, value, expVal, txn)
+}
+
+// MVCCBlindConditionalPut is a fast-path of MVCCConditionalPut. See the
+// MVCCConditionalPut comments for details of the
+// semantics. MVCCBlindConditionalPut skips retrieving the existing metadata
+// for the key requiring the caller to guarantee no versions for the key
+// currently exist.
+func MVCCBlindConditionalPut(
```

Comment:
If you do a `ConditionalPut` which expects the value to exist you could use `MVCCBlindConditionalPut` to generate the appropriate error. This seems preferable to calling either `MVCCConditionalPut` (which is slower) or moving the error generation out of the `storage/engine` directory.